wavefront_python_sdk/direct_ingestion/wavefront_direct_ingestion_client.py

```diff
@@ -0,0 +1,115 @@
+from __future__ import absolute_import, division, print_function
+
+from wavefront_python_sdk.common.connection_handler import IConnectionHandler
+from wavefront_python_sdk.common.utils import metric_to_line_data, \
+    histogram_to_line_data, tracing_span_to_line_data
+
+import sys
+import requests
+from threading import Timer
+from queue import Queue
+
+
+class WavefrontDirectIngestionClient(IConnectionHandler):
+    WAVEFRONT_METRIC_FORMAT = 'wavefront'
+    WAVEFRONT_HISTOGRAM_FORMAT = 'histogram'
+    WAVEFRONT_TRACING_SPAN_FORMAT = 'trace'
+
+    def __init__(self, server, token, max_queue_size=50000, batch_size=10000, flush_interval_seconds=1):
+        IConnectionHandler.__init__(self)
+        self.server = server
+        self._token = token
+        self._max_queue_size = max_queue_size
+        self._batch_size = batch_size
+        self._flush_interval_seconds = flush_interval_seconds
+        self._default_source = "wavefrontDirectSender"
+        self._metrics_buffer = Queue(max_queue_size)
+        self._histograms_buffer = Queue(max_queue_size)
+        self._tracing_spans_buffer = Queue(max_queue_size)
+        self._headers = {'Content-Type': 'text/plain', 'Authorization': 'Bearer ' + token}
+        self._flush()
+
+    def _report(self, points, data_format):
+        try:
+            params = {'f': data_format}
+            r = requests.post(self.server + '/report', params=params, headers=self._headers, data=points)
```

Comment:
There is a ticket to add gzip support for python. That would result in adding `Content-Encoding: gzip` and `Content-Length` to the headers at report time. I believe the requests module may support it directly in which case we may just need to set some flag during the request.

Reply:
Already solved that, haven't push yet